File: samples/binarySearch.js
function sortedIndex(haystack, needle) {
  var low = 0,
      high = haystack ? haystack.length : low,
      current;

  while (low < high) {
    current = (low + high) >>> 1;
    if (haystack[current] < needle) {
      low = current + 1;
    } else {
      high = current;
    }
  }

  return low;
}

File: samples/binary_search.rb
def binary_search(haystack, needle)
  low = 0
  high = haystack ? haystack.length : low
  current = nil
  while low < high
    current = low + high >>> 1
    if haystack.current < needle
      low = current + 1
    else
      high = current
    end
  return low
end
